ntoskrnl/io/pnpmgr/pnpreport.c

```diff
@@ -464,9 +464,9 @@ IoReportTargetDeviceChange(IN PDEVICE_OBJECT PhysicalDeviceObject,
     ASSERT(notifyStruct->FileObject == NULL);
 
     /* Do not handle system PnP events */
-    if ((RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_QUERY_REMOVE), sizeof(GUID)) != sizeof(GUID)) ||
-        (RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_CANCELLED), sizeof(GUID)) != sizeof(GUID)) ||
-        (RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_COMPLETE), sizeof(GUID)) != sizeof(GUID)))
+    if ((RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_QUERY_REMOVE), sizeof(GUID)) == sizeof(GUID)) ||
+        (RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_CANCELLED), sizeof(GUID)) == sizeof(GUID)) ||
+        (RtlCompareMemory(&(notifyStruct->Event), &(GUID_TARGET_DEVICE_REMOVE_COMPLETE), sizeof(GUID)) == sizeof(GUID)))
```

Comment:
This change looks a bit sus to me, needs to be inspected carefully.

Reply:
From MSDN doc:

>Return value

>RtlCompareMemory returns the number of bytes in the two blocks that match. If all bytes match up to the specified >Length value, the Length value is returned.

>Remarks

>The routine starts by comparing the first byte in the first block to the first byte in the second block, and continues >to compare successive bytes in the two blocks while the bytes match. The routine stops comparing bytes when it >encounters the first pair of bytes that are not equal, or when the number of matching bytes equals the Length >parameter value, whichever occurs first.

My understanding is, if we don't want to handle those notifications here, then this is a correct thing to do.
Eventually a testbot run on this PR would be nice and interesting though, if the test suite can stress it.

Reply:
There is `IsEqualGUID` function for that